test/parallel/test-fs-realpath-pipe.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const common = require('../common');
+
+if (common.isWindows) {
+  common.skip('This test does not apply to Windows.');
+  return;
+}
+
+const assert = require('assert');
+
+const expected = '/dev/stdin';
+
+const { spawnSync } = require('child_process');
+
+for (const code of [
+  `require('fs').realpath('/dev/stdin', (err, resolvedPath) => {
+    if (err) {
+      process.exit(1);
+    }
+  });`,
+  `try {
+    require('fs').realpathSync('/dev/stdin');
+  } catch (e) {
+    process.exit(1);
+  };`
+]) {
+  assert.strictEqual(spawnSync(process.execPath, ['-e', code], {
```

Comment:
In adition to sanity, I would consider asserting the value of `resolvedPath` and the returned value from `require('fs').realpathSync('/dev/stdin');`
Either in the script literal or `console.log`ing it and adding another assert for stdout.